src/engine/runtime.js

```diff
@@ -658,12 +685,35 @@ class Runtime extends EventEmitter {
         this._pushMonitors();
         const doneThreads = this.sequencer.stepThreads();
         this._updateGlows(doneThreads);
-        this._setThreadCount(this.threads.length + doneThreads.length);
+        // Add done threads so that even if a thread finishes within 1 frame, the green
+        // flag will still indicate that a script ran.
+        this._emitProjectRunStatus(
+            this.threads.length + doneThreads.length -
+                this._getMonitorThreadCount([...this.threads, ...doneThreads]));
         if (this.renderer) {
             // @todo: Only render when this.redrawRequested or clones rendered.
             this.renderer.draw();
         }
         if (this._refreshTargets) this.emit(Runtime.TARGETS_UPDATE);
+
+        // @todo(vm#570) only emit if monitors has changed since last time.
```

Comment:
Maybe we should think about moving to immutable objects for state, so we can get this for free everywhere we need it? https://facebook.github.io/immutable-js/